src/hooks/baseHttp.ts

```diff
@@ -0,0 +1,387 @@
+          truncated = true;
+          chunk = chunk.substr(0, allowedLengthToAdd);
+        }
+        body += chunk;
+      }
+      if (args[0] === 'end') {
+        const responseData = {
+          statusCode,
+          receivedTime,
+          body,
+          headers: lowerCaseObjectKeys(headers),
+        };
+        const httpSpan = getHttpSpan(
+          transactionId,
+          awsRequestId,
+          requestRandomId,
+          requestData,
+          Object.assign({ truncated }, responseData)
+        );
+        if (httpSpan.id !== requestRandomId) {
+          // In Http case, one of our parser decide to change the spanId for async connection
+          SpansContainer.changeSpanId(requestRandomId, httpSpan.id);
+        }
+        SpansContainer.addSpan(httpSpan);
+      }
+      GlobalDurationTimer.stop();
+    };
+  }
+
+  static isBlacklisted(host: string) {
+    return host === getEdgeHost() || hostBlaclist.has(host);
+  }
+
+  static aggregateRequestBodyToSpan(
+    body,
+    requestData: RequestData,
+    currentSpan: BasicChildSpan,
+    maxSize: number = getEventEntitySize(true)
+  ) {
+    let serviceData: ServiceData = {};
+    if (body && !requestData.truncated) {
+      requestData.body += body;
+      serviceData = getServiceData(requestData, null);
+      const truncated = maxSize < requestData.body.length;
+      if (truncated) requestData.body = requestData.body.substr(0, maxSize);
+      requestData.truncated = truncated;
+    }
+    if (currentSpan) {
+      // @ts-ignore
+      currentSpan.info.httpInfo = getHttpInfo(requestData, {});
+      Object.assign(currentSpan.info, {
+        ...serviceData.awsServiceData,
+      });
+    }
+  }
+
+  @GlobalDurationTimer.timedSync()
+  static parseHttpRequestOptions(options: ParseHttpRequestOptions = {}, url?: string): RequestData {
+    const host = BaseHttp._getHostFromOptionsOrUrl({ options, url });
+    const agent = options.agent || options._defaultAgent;
+
+    let path = null;
+    let port = null;
+    let search = null;
+    let protocol = null;
+
+    const { headers, method = 'GET' } = options;
+    const sendTime = new Date().getTime();
+
+    if (url) {
+      const myUrl = new URL(url);
+      ({ pathname: path, port, protocol, search } = myUrl);
+    } else {
+      path = options.path || '/';
+      port = options.port || options.defaultPort || (agent && agent.defaultPort) || 80;
+      protocol = options.protocol || (port === 443 && 'https:') || 'http:';
+    }
+
+    const modifiedHeaders = headers && !headers.host ? addHeaders(headers, { host }) : headers;
+
+    if (!!search) {
+      search = BaseHttp.scrubQueryParams(search);
+    } else {
+      search = '';
+    }
+
+    const uri = `${host}${path}${search}`;
+
+    return {
+      truncated: false,
+      path,
+      port,
+      uri,
+      host,
+      body: '', // XXX Filled by the httpRequestEndWrapper or httpRequestEmitBeforeHookWrapper ( / Write)
```

Comment:
Is XXX placeholder for something?
```suggestion
      body: '', // XXX Filled by the httpRequestEndWrapper or httpRequestEmitBeforeHookWrapper ( / Write)
```

Reply:
IDK this was directly copied from the previous implementation...
Anyways removed the XXX and kept the rest of the comment